### PR TITLE
AST-2096 - Changed Border radius control from slider to inputs values

### PR DESCRIPTION
**Pull request by er-roshan**
### Description
AST-2096 - Changed Border radius control from slider to inputs values

### Screenshots
https://d.pr/i/HPfo1H

### Types of changes
New feature (non-breaking change which adds functionality)
Added new controls for border radius

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: assets/js/unminified/customizer-preview.js [hunk 1]
```diff
@@ -915,6 +915,43 @@ function hasWordPressWidgetBlockEditor() {
 		} );
 	} );
 
+	/**
+	 * Button Border Radius Fields
+	 */
+	wp.customize( 'astra-settings[button-radius-fields]', function( setting ) {
```
Comment (imnavanath) [inline assets/js/unminified/customizer-preview.js:921]: As we migrated old control to new, why we are maintaining old code, it increases our lines of code. See L-900 above same file. Replace this over there.

File: inc/builder/type/base/assets/js/customizer-preview.js [hunk 2]
```diff
@@ -200,6 +200,25 @@ function astra_builder_button_css( builder_type = 'header', button_count ) {
 		astra_builder_visibility_css( section, selector, 'block' );
 
 		(function (index) {
+			// Builder Type Border Radius Fields
+			wp.customize('astra-settings[' + builder_type + '-button' + index + '-border-radius-fields]', function (setting) {
```
Comment (imnavanath) [inline inc/builder/type/base/assets/js/customizer-preview.js:204]: Same as above, verify once.

File: inc/builder/type/base/assets/js/customizer-preview.js [hunk 3]
```diff
@@ -344,6 +363,27 @@ function astra_builder_social_css( builder_type = 'header', social_count ) {
 			'px'
 		);
 
+		// Icon Border Radius Fields
+		wp.customize('astra-settings[' + builder_type + '-social-' + index + '-radius-fields]', function (setting) {
```
Comment (imnavanath) [inline inc/builder/type/base/assets/js/customizer-preview.js:367]: Same as above, verify once.

File: inc/builder/type/base/dynamic-css/social/class-astra-social-component-dynamic-css.php [hunk 4]
```diff
@@ -45,10 +45,11 @@ public static function astra_social_dynamic_css( $builder_type = 'header' ) {
...
 			$_section        = ( 'header' === $builder_type ) ? 'section-hb-social-icons-' . $index : 'section-fb-social-icons-' . $index;
 
-			$icon_spacing    = astra_get_option( $builder_type . '-social-' . $index . '-space' );
-			$icon_bg_spacing = astra_get_option( $builder_type . '-social-' . $index . '-bg-space' );
-			$icon_size       = astra_get_option( $builder_type . '-social-' . $index . '-size' );
-			$icon_radius     = astra_get_option( $builder_type . '-social-' . $index . '-radius' );
+			$icon_spacing       = astra_get_option( $builder_type . '-social-' . $index . '-space' );
+			$icon_bg_spacing    = astra_get_option( $builder_type . '-social-' . $index . '-bg-space' );
+			$icon_size          = astra_get_option( $builder_type . '-social-' . $index . '-size' );
+			$icon_radius        = astra_get_option( $builder_type . '-social-' . $index . '-radius' );
```
Comment (imnavanath) [inline inc/builder/type/base/dynamic-css/social/class-astra-social-component-dynamic-css.php:51]: Is this something we deprecated right?

File: inc/builder/type/header/menu/assets/js/unminified/customizer-preview.js [hunk 5]
```diff
@@ -360,6 +360,32 @@
 					} );
 				} );
 
+				// Sub Menu - Border Radius Fields.
+				wp.customize( 'astra-settings[header-menu'+ index +'-submenu-border-radius-fields]', function( value ) {
```
Comment (imnavanath) [inline inc/builder/type/header/menu/assets/js/unminified/customizer-preview.js:364]: Same as above, verify once. if deprecated control's preview is there then replace this over there.

File: inc/builder/type/base/dynamic-css/button/class-astra-button-component-dynamic-css.php [hunk 6]
```diff
@@ -48,10 +48,12 @@ public static function astra_button_dynamic_css( $builder_type = 'header' ) {
...
 			$_prefix = 'button' . $index;
 
-			$selector             = '.ast-' . $builder_type . '-button-' . $index;
-			$button_font_size     = astra_get_option( $builder_type . '-' . $_prefix . '-font-size' );
-			$button_border_width  = astra_get_option( $builder_type . '-' . $_prefix . '-border-size' );
-			$button_border_radius = astra_get_option( $builder_type . '-' . $_prefix . '-border-radius' );
+			$selector                    = '.ast-' . $builder_type . '-button-' . $index;
+			$button_font_size            = astra_get_option( $builder_type . '-' . $_prefix . '-font-size' );
+			$button_border_width         = astra_get_option( $builder_type . '-' . $_prefix . '-border-size' );
+			$button_border_radius        = astra_get_option( $builder_type . '-' . $_prefix . '-border-radius' );
```
Comment (imnavanath) [inline inc/builder/type/base/dynamic-css/button/class-astra-button-component-dynamic-css.php:54]: Is this something we deprecated right?

File: inc/builder/type/header/mobile-trigger/assets/js/unminified/customizer-preview.js [hunk 7]
```diff
@@ -63,6 +63,30 @@
 		} );
 	} );
 
+	// Border Radius Fields.
+	wp.customize( 'astra-settings[mobile-header-toggle-border-radius-fields]', function( setting ) {
```
Comment (imnavanath) [inline inc/builder/type/header/mobile-trigger/assets/js/unminified/customizer-preview.js:67]: Same as above. also check all instances previously done.

File: inc/builder/type/header/woo-cart/assets/js/unminified/customizer-preview.js [hunk 8]
```diff
@@ -272,6 +272,32 @@
 		});
 	});
 
+	/**
+	 * Cart Border Radius Fields
+	 */
+	wp.customize('astra-settings[woo-header-cart-icon-radius-fields]', function (setting) {
```
Comment (imnavanath) [inline inc/builder/type/header/woo-cart/assets/js/unminified/customizer-preview.js:278]: same as above

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php [hunk 9]
```diff
@@ -37,6 +37,7 @@ function astra_hb_woo_cart_dynamic_css( $dynamic_css, $dynamic_css_filtered = ''
 	$icon_hover_color      = esc_attr( astra_get_option( 'header-woo-cart-icon-hover-color' ) ); // icon cart hover color.
 
 	$header_cart_icon_radius         = astra_get_option( 'woo-header-cart-icon-radius' );
```
Comment (imnavanath) [inline inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:39]: Is this something we deprecated right?

File: inc/class-astra-dynamic-css.php [hunk 10]
```diff
@@ -251,9 +251,10 @@ public static function return_output( $dynamic_css, $dynamic_css_filtered = '' )
 			}
 
 			// Button Styling.
-			$btn_border_radius     = astra_get_option( 'button-radius' );
-			$theme_btn_padding     = astra_get_option( 'theme-button-padding' );
-			$highlight_theme_color = astra_get_foreground_color( $theme_color );
+			$btn_border_radius        = astra_get_option( 'button-radius' );
```
Comment (imnavanath) [inline inc/class-astra-dynamic-css.php:254]: Please check all the deprecated cases, remove extra code which will no longer required. 
Check following files as well, I stopped review here.

File: assets/js/unminified/customizer-preview.js (hunk 1 above)
Comment (er-roshan) [reply assets/js/unminified/customizer-preview.js:900]: @imnavanath I have removed the old code.

File: inc/builder/type/base/assets/js/customizer-preview.js (hunk 2 above)
Comment (er-roshan) [reply inc/builder/type/base/assets/js/customizer-preview.js:196]: @imnavanath I have removed the old code.

File: inc/builder/type/base/assets/js/customizer-preview.js (hunk 3 above)
Comment (er-roshan) [reply inc/builder/type/base/assets/js/customizer-preview.js:351]: I have removed the old code.

File: inc/builder/type/header/menu/assets/js/unminified/customizer-preview.js (hunk 5 above)
Comment (er-roshan) [reply inc/builder/type/header/menu/assets/js/unminified/customizer-preview.js:351]: I have removed the old code.

File: inc/builder/type/base/dynamic-css/button/class-astra-button-component-dynamic-css.php (hunk 6 above)
Comment (er-roshan) [reply inc/builder/type/base/dynamic-css/button/class-astra-button-component-dynamic-css.php]: I have removed the old code.

File: inc/builder/type/header/mobile-trigger/assets/js/unminified/customizer-preview.js (hunk 7 above)
Comment (er-roshan) [reply inc/builder/type/header/mobile-trigger/assets/js/unminified/customizer-preview.js:67]: I have removed the old code.

File: inc/builder/type/header/woo-cart/assets/js/unminified/customizer-preview.js (hunk 8 above)
Comment (er-roshan) [reply inc/builder/type/header/woo-cart/assets/js/unminified/customizer-preview.js:268]: I have removed the old code.

**Review by yssambare12 — COMMENTED** [review]
@er-roshan Update the branch with the latest changes.

**Review by neerajmasai — APPROVED** [review]

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php [hunk 11]
```diff
@@ -36,7 +36,7 @@ function astra_hb_woo_cart_dynamic_css( $dynamic_css, $dynamic_css_filtered = ''
 	$icon_color            = esc_attr( astra_get_option( 'header-woo-cart-icon-color', $theme_color ) );
 	$icon_hover_color      = esc_attr( astra_get_option( 'header-woo-cart-icon-hover-color' ) ); // icon cart hover color.
 
-	$header_cart_icon_radius         = astra_get_option( 'woo-header-cart-icon-radius' );
+	$header_cart_icon_radius_fields  = astra_get_option( 'woo-header-cart-icon-radius-fields' );
```
Comment (imnavanath) [inline inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:39]: Not getting its default migration code! @er-roshan can you verify this once.

File: inc/core/class-astra-theme-options.php [hunk 12]
```diff
@@ -211,6 +211,29 @@ public static function defaults() {
 						'mobile-unit'  => 'px',
 					),
 					'button-radius'                        => 2,
```
Comment (imnavanath) [inline inc/core/class-astra-theme-options.php:213]: Can be remove??

Comment (er-roshan) [reply inc/core/class-astra-theme-options.php]: I have removed the line

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php (hunk 11 above)
Comment (er-roshan) [reply inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:39]: @imnavanath I have added the missing defaults for **woo-header-cart-icon-radius-fields**.

**Review by yssambare12 — COMMENTED** [review]
@er-roshan please check E2E test, it's failing.

**Comment by imnavanath** [comment]
@er-roshan Test cases e2e pipeline failed, you can either resolve that border-radius test cases -
1. `specs/customizer/footer-builder/elements/social-icon/icon-radius.test.js`
2. `specs/customizer/header-builder/elements/social-icons/social-icon-radius.test.js`

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php [hunk 13]
```diff
@@ -155,14 +152,17 @@ function astra_hb_menu_dynamic_css( $dynamic_css, $dynamic_css_filtered = '' ) {
...
+				'border-bottom-width'        => astra_get_css_value( $sub_menu_border_bottom, 'px' ),
+				'border-right-width'         => astra_get_css_value( $sub_menu_border_right, 'px' ),
+				'border-left-width'          => astra_get_css_value( $sub_menu_border_left, 'px' ),
+				'border-color'               => esc_attr( astra_get_option( 'header-' . $_prefix . '-submenu-b-color', $theme_color ) ),
+				'border-style'               => 'solid',
+				'width'                      => astra_get_css_value( $sub_menu_width, 'px' ),
+				'border-top-left-radius'     => astra_responsive_spacing( $sub_menu_border_radius_fields, 'top', 'desktop' ),
+				'border-top-right-radius'    => astra_responsive_spacing( $sub_menu_border_radius_fields, 'right', 'desktop' ),
+				'border-bottom-right-radius' => astra_responsive_spacing( $sub_menu_border_radius_fields, 'bottom', 'desktop' ),
+				'border-bottom-left-radius'  => astra_responsive_spacing( $sub_menu_border_radius_fields, 'left', 'desktop' ),
```
Comment (imnavanath) [inline inc/builder/type/header/menu/dynamic-css/dynamic.css.php:165]: It's tablet, mobile dynamic CSS code missing in this file @er-roshan??

Comment (er-roshan) [reply inc/builder/type/header/menu/dynamic-css/dynamic.css.php:165]: @imnavanath I have resolved the missing css

**Review by yssambare12 — APPROVED** [review]

**Review by imnavanath — APPROVED** [review]

**Pull request merged**